src/core/manager.c

```diff
@@ -3801,6 +3801,12 @@ int manager_reload(Manager *m) {
         (void) manager_setup_handoff_timestamp_fd(m);
         (void) manager_setup_pidref_transport_fd(m);
 
+        // XXX: Clean up deserialized bus track information. They're never consumed during reload (as opposed
+        // to reexec) since we do not disconnect from the bus. manager_(de)serialize() should probably
+        // distinguish reload/reexec properly in the first place.
```

Comment:
tbh i kinda would prefer to keep the differences between reload and reexec on the deserialization side rather than on the serialization side. i think it's a nice property if the serialization is independent from the intended use.

Hence, I would probably just add an explanatory comment here, and drop the FIXME one, but do exactly what your patch does here codewise.